entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverter.java

```diff
@@ -65,24 +62,14 @@ public FilterTypeExpression convert(
 
     switch (operator) {
       case EQ:
-        // 'field' EQ 'null' -> 'field' EQ 'null' || 'field' NOT_EXISTS
-        RelationalExpression nonExistenceRelationalExpression =
-            RelationalExpression.of(
-                identifierExpression, convertOperator(Operator.NOT_EXISTS), constantExpression);
-        return LogicalExpression.builder()
-            .operator(LogicalOperator.OR)
-            .operands(List.of(nonExistenceRelationalExpression, relationalExpression))
-            .build();
+        // 'field' EQ 'null' -> 'field' NOT_EXISTS
+        return RelationalExpression.of(
+            identifierExpression, convertOperator(Operator.NOT_EXISTS), constantExpression);
 
       case NEQ:
-        // 'field' NEQ 'null' -> 'field' NEQ 'null' && 'field' EXISTS
-        RelationalExpression existenceRelationalExpression =
-            RelationalExpression.of(
-                identifierExpression, convertOperator(Operator.EXISTS), constantExpression);
-        return LogicalExpression.builder()
-            .operator(LogicalOperator.AND)
-            .operands(List.of(existenceRelationalExpression, relationalExpression))
-            .build();
+        // 'field' NEQ 'null' -> 'field' EXISTS
```

Comment:
so the assumption here is that if a field is set, it cant be null?

Reply:
right